flow/project.py

```diff
@@ -2664,6 +2690,20 @@ def compute_status(data):
                 file=err,
             )
 
+        def combine_group_and_operation_status(aggregate_status_results):
+            group_statuses = {}
+            for status_result in aggregate_status_results:
+                group_name = status_result["group_name"]
+                # Only true when both a user group and singleton group report a status. We use the
+                # one that is not unknown.
+                if (
+                    group_name in group_statuses
+                    and status_result["status"]["scheduler_status"] == JobStatus.unknown
+                ):
+                    continue
```

Comment:
Could this be flipped to `break` when the one that is not unknown is found? If there is only one, we should break after finding it, rather than `continue` on all the other cases, to avoid doing more iterations than needed. I can't quite wrap my head around whether this is equivalent or not, without testing this manually and examining the data structures.

Reply:
The break is not equivalent as we are iterating over the status with multiple operations. This check just ensures that we don't overwrite the `group_statuses` dict when the status from a user group or singleton group would overwrite the status of the other with a `JobStatus.unknown` potentially losing information.